helpdesk_mgmt_timesheet_time_control/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 Graeme Gellatly
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class HelpdeskTicket(models.Model):
+    _name = "helpdesk.ticket"
+    _inherit = ["helpdesk.ticket", "hr.timesheet.time_control.mixin"]
+
+    @api.model
+    def _relation_with_timesheet_line(self):
+        return "ticket_id"
+
+    @api.depends(
+        "team_id.allow_timesheet",
+        "project_id.allow_timesheets",
+        "timesheet_ids.employee_id",
+        "timesheet_ids.unit_amount",
+    )
+    def _compute_show_time_control(self):
+        result = super()._compute_show_time_control()
+        for ticket in self:
+            # Never show button if timesheets are not allowed in project
+            if not (
+                ticket.project_id.allow_timesheets and ticket.team_id.allow_timesheet
+            ):
+                ticket.show_time_control = False
```

Comment:
```suggestion
            ticket.show_time_control = ticket.project_id.allow_timesheets and ticket.team_id.allow_timesheet
```

Reply:
@Yajo this field is not a Boolean so we can't do this.